crates/feldera-types/src/format/json.rs

```diff
@@ -51,6 +51,29 @@ pub struct JsonParserConfig {
     /// [{"b": true, "i": 0},{"b": false, "i": 100, "s": "foo"}]
     /// ```
     pub array: bool,
+
+    /// Whether JSON elements can span multiple lines.
+    ///
+    /// This only affects JSON input.
+    pub lines: JsonLines,
+}
+
+/// Whether JSON elements (objects or arrays) can span multiple lines.
+#[derive(Deserialize, Serialize, Clone, Debug, Default, PartialEq, Eq, ToSchema)]
+pub enum JsonLines {
+    /// A given JSON element never contains a new-line..
+    ///
+    /// Suitable for parsing [NDJSON](https://github.com/ndjson/ndjson-spec).
+    /// Lines are allowed to contain multiple JSON elements.
+    ///
+    /// The parser ignores empty lines.
+    #[default]
+    Single,
+
+    /// Individual JSON elements may span multiple lines.
+    ///
+    /// This setting can also parse all the same formats as [JsonLines::Single].
+    Multiple,
```

Comment:
Let's add `serde(rename="single")`, `serde(rename="multiple")` for consistency with other config options.

Reply:
`Multiple` might be a better default, since it's more permissive and I don't expect the performance overhead to be very significant. What are the downsides?

Reply:
Probably not much. If you want, I can just drop the setting entirely in favor of always supporting multiline JSON.

Reply:
Fewer knows is always better

Reply:
Agreed, let's default to Multiple.

Reply:
It might be a good idea to keep the option. When you know your input is ndjson, `lines=single` might produce more useful errors.

Reply:
I fixed all of these.

Reply:
> It might be a good idea to keep the option. When you know your input is ndjson, `lines=single` might produce more useful errors.

I had the same conclusion.